node/utils/metrics/organization.ts

```diff
@@ -0,0 +1,96 @@
+import type { Logger } from '@vtex/api/lib/service/logger/logger'
+
+import type { Metric } from './metrics'
+import { sendMetric } from './metrics'
+
+interface UpdateOrganizationFieldsMetric {
+  update_details: { properties: string[] }
+}
+
+type UpdateOrganization = Metric & {
+  fields: UpdateOrganizationFieldsMetric
+}
+
+export interface UpdateOrganizationParams {
+  account: string
+  currentOrganizationData: Organization
+  updatedProperties: UpdatedOrganizationProps
+}
+
+const buildUpdateOrganizationMetric = (
+  account: string,
+  updatedProperties: string[]
+): UpdateOrganization => {
+  const updateOrganizationFields: UpdateOrganizationFieldsMetric = {
+    update_details: { properties: updatedProperties },
+  }
+
+  return {
+    account,
+    description: 'Update Organization Action - Graphql',
+    fields: updateOrganizationFields,
+    kind: 'update-organization-graphql-event',
+    name: 'b2b-suite-buyerorg-data',
+  } as UpdateOrganization
+}
+
+const getFieldsNamesByFieldsUpdated = (
+  updateOrganizationParams: UpdateOrganizationParams
+): string[] => {
+  const updatedPropName: string[] = []
+
+  const { currentOrganizationData, updatedProperties } =
+    updateOrganizationParams
+
+  for (const key in updatedProperties) {
+    if (Object.prototype.hasOwnProperty.call(updatedProperties, key)) {
+      const value = updatedProperties[key as keyof typeof updatedProperties]
+
+      // I tried to compare the objects value !== currentOrganizationData[key as keyof Organization,
+      // but it was not working, so I use JSON.stringify
+      if (
+        JSON.stringify(value) !==
+        JSON.stringify(currentOrganizationData[key as keyof Organization])
```

Comment:
This kind of comparison is kind of expensive, depending on the size of the objects being stringified. It's also not guaranteed to work, as stringified objects don't have any guarantees regarding the order of keys, for example. So the objects might be the same, but the stringified versions might not match due to sorting.

Do we really need to do this level of comparison? If it's really essential, it might be better to analyze the fields that can be updated and see how complex they are, to see if we can do simpler comparisons for different types of fields.

Reply:
Yes, we need to make a diff between the persisted info and the changed. Do you have a suggestion to simplify?

Reply:
It depends on the contents of that object. The more nested it is, the more complex it will be. Do we know what keys may be on the object and what they're contents are? If you can grab a good example, as complete as possible, we can analyse it better to see the best way to do the comparison.

Reply:
The example is the organization structure
```Typescript
interface Organization {
  id: string
  name: string
  tradeName?: string
  collections: Collection[]
  costCenters: string[]
  paymentTerms: PaymentTerm[]
  priceTables?: Price[]
  status: string
  created: string
  customFields?: CustomField[]
  salesChannel?: string
  sellers?: Seller[]
}
```

Reply:
What do you think to use [`isEqual` from `lodash`](https://www.npmjs.com/package/@types/lodash.isequal)?

Reply:
@Rudge isEqual would be more trustworthy than the JSON.stringify, as it won't fail due to sorting of keys.

It will still be expensive though, so I'd prefer avoiding if possible. At least this happens on the background though, so we won't delay the response by doing it, just spend more compute resources.

Do you have real examples of an update object and the original organization object? The types are not very trustworthy, they may be missing fields or being simpler than the real thing. With a real example this can be more tangible, so that we have a better idea of the sizes as well as the type of comparisons that will be made.

Reply:
A real example is it.
```JSON
{
    "id": "3956085d-36b8-11ee-83ab-0e170ba3b9a5",
    "name": "BAHIA EVENTOS E LOCAÇÕES LTDA.",
    "tradeName": "BA EVENTOS",
    "status": "active",
    "created": "2023-08-09T13:25:36.387Z",
    "collections": [],
    "paymentTerms": [
        {
            "id": "6",
            "name": "Boleto Bancário",
        },
        {
            "id": "999999",
            "name": "Credit card",
        },
        {
            "id": "125",
            "name": "Pix"
        },
        {
            "id": "172",
            "name": "Pagaleve"
        }
    ],
    "priceTables": [
        "pjba"
    ],
    "salesChannel": null,
    "costCenters": [],
    "sellers": null,
    "customFields": []
}
```

Reply:
Esse aí é qual dos dois objetos? E pode colocar o outro também (se esse for a org original, colar o obj de update, ou vice-versa). Peço porque muitas vezes esses updates são bem mais simples do que a estrutura completa do objeto.

Reply:
Mas pelo que tou vendo me parece um objeto bem simples. Com isEqual do lodash ficaria uma comparação mais garantida, e por ser um objeto simples não vai ficar tão caro. Rodando em background como está é melhor ainda. Me parece ok, fico mais tranquila assim!

Reply:
Esse é o objeto que é enviado para atualizar e contem a mesma estrutura do objeto original. Lembrando que toda vez que muda um campo na organização a estrutura é enviada inteiramente e cada estrutura dessa é uma aba ou uma seção dessa tela.
![image](https://github.com/vtex-apps/b2b-organizations-graphql/assets/5332361/e0de7d4b-5604-44de-aa5a-3072a0cdf989)